### ISSUE TITLE
Evaluate why :preferences strategy does not work in Xcode 7

### ISSUE THREAD

**Issue by jmoody** [calabash/calabash-ios#860]
I used this script in Instruments.app > Automation:

```
var target = UIATarget.localTarget();

target.frontMostApp().setPreferencesValueForKey("value", "com.my.key")

var myValue  = target.frontMostApp().preferencesValueForKey("com.my.key")
UIALogger.logPass("myValue: ".concat(myValue));
```

and verified that the UIAutomation Preferences API is working in Xcode 7 _for both simulators and devices_.  Previously, it was only working for simulators.

The next step is to check which plist is being updated on the iOS Simulator.

@krukow @sapieneptus @john7doe @rasmuskl

**Comment by john7doe** [comment]
Perhaps you can use diff -r to find it:

Do a copy of the simulator before and after you update the prefs and
compare the folder with diff -r

Br,
/Simon


On Mon, Sep 21, 2015 at 11:30 AM Joshua Moody <notifications@github.com>
wrote:

> I used this script in Instruments.app > Automation:
>
> var target = UIATarget.localTarget();
>
> target.frontMostApp().setPreferencesValueForKey("value", "com.my.key")
>
> var myValue  = target.frontMostApp().preferencesValueForKey("com.my.key")
> UIALogger.logPass("myValue: ".concat(myValue));
>
> and verified that the UIAutomation Preferences API is working in Xcode 7 *for
> both simulators and devices*. Previously, it was only working for
> simulators.
>
> The next step is to check which plist is being updated on the iOS
> Simulator.
>
> @krukow <https://github.com/krukow> @sapieneptus
> <https://github.com/sapieneptus> @john7doe <https://github.com/john7doe>
> @rasmuskl <https://github.com/rasmuskl>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/calabash/calabash-ios/issues/860>.
>
-- 
Join us at Xamarin Evolve <http://evolve.xamarin.com>, the world's largest
cross-platform mobile development event, April 24-28, 2016!

**Comment by jmoody** [comment]
My trick is put the whole simulator into a temporary git repo and view the diff.

I also discovered this awesome tool:

```
# Available with brew.
$ watch --interval 0.1 --differences cat device.plist 
```

**Comment by sapieneptus** [comment]
My investigation from before indicates that the path is unchanged from the previous release. 

The problem _appeared_ to be that `[NSDictionary writeToFile]` generates a plaintext plist, whereas `[NSUserDefaults setObject:forKey]` encodes it as a binary plist.

**Comment by sapieneptus** [comment]
@jmoody If you'd like to see it in action, you can try my branch of the server: https://github.com/calabash/calabash-ios-server/tree/feature/make-preferences-work

**Comment by jmoody** [comment]
Added support in the server with **Fix :preferences strategy for iOS 9** [#267](https://github.com/calabash/calabash-ios-server/pull/267)

:preferences will be available in 0.17.0.
